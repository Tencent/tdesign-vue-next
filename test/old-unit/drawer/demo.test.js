/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import attachParentVue from '@/examples/drawer/demos/attach-parent.vue';
import baseVue from '@/examples/drawer/demos/base.vue';
import customVue from '@/examples/drawer/demos/custom.vue';
import destroyVue from '@/examples/drawer/demos/destroy.vue';
import noMaskVue from '@/examples/drawer/demos/no-mask.vue';
import operationVue from '@/examples/drawer/demos/operation.vue';
import placementVue from '@/examples/drawer/demos/placement.vue';
import popupVue from '@/examples/drawer/demos/popup.vue';
import sizeVue from '@/examples/drawer/demos/size.vue';

const mapper = {
  attachParentVue,
  baseVue,
  customVue,
  destroyVue,
  noMaskVue,
  operationVue,
  placementVue,
  popupVue,
  sizeVue,
};

describe('Drawer', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Drawer ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
