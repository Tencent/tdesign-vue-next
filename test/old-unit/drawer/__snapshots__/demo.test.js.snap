// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Drawer attachParentVue demo works fine 1`] = `
<div
  class="t-container"
>
  <div
    class="t-suf-container"
  >
    <div
      class="t-drawer t-drawer--right t-drawer--attach"
    >
      <div
        class="t-drawer__mask"
      />
      <div
        class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
        style="width: 300px;"
      >
        <div
          class="t-drawer__header"
        >
          抽屉标题
        </div>
        <!---->
        <div
          class="t-drawer__body narrow-scrollbar"
        >
          
          <p>
            This is a drawer
          </p>
          
        </div>
        <div
          class="t-drawer__footer"
        >
          <div
            style="display: flex; justify-content: flex-start;"
          >
            <button
              class="t-button t-size-m t-button--variant-base t-button--theme-primary t-drawer__confirm"
              type="button"
            >
              <span
                class="t-button__text"
              >
                确认
              </span>
            </button>
            <button
              class="t-button t-size-m t-button--variant-base t-button--theme-default t-drawer__cancel"
              type="button"
            >
              <span
                class="t-button__text"
              >
                取消
              </span>
            </button>
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div
      class="tdesign-radio-button"
    >
       抽屉弹出方向： 
      <div
        class="t-radio-group t-size-m t-radio-group__outline"
      >
        
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="left"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
             left 
            
            
          </span>
        </label>
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="right"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
             right 
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="top"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
             top 
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="bottom"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
             bottom 
            
            
          </span>
        </label>
        
        <!---->
      </div>
    </div>
    <br />
    <div
      class="tdesign-radio-button"
    >
       抽屉弹出模式： 
      <div
        class="t-radio-group t-size-m t-radio-group__outline"
      >
        
        <label
          class="t-radio-button t-is-checked"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="overlay"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
             overlay 
            
            
          </span>
        </label>
        <label
          class="t-radio-button"
        >
          <input
            class="t-radio-button__former"
            name=""
            type="radio"
            value="push"
          />
          <span
            class="t-radio-button__input"
          />
          <span
            class="t-radio-button__label"
          >
            
            
             push 
            
            
          </span>
        </label>
        
        <!---->
      </div>
    </div>
    <br />
    <br />
    <button
      class="t-button t-size-m t-button--variant-outline t-button--theme-default"
      type="button"
    >
      <span
        class="t-button__text"
      >
        
        打开抽屉
        
      </span>
    </button>
  </div>
</div>
`;

exports[`Drawer Drawer baseVue demo works fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer--right"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        header
      </div>
      <div
        class="t-drawer__close-btn"
      >
        <svg
          class="t-icon t-icon-close t-submenu-icon"
          fill="none"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
        >
          <path
            d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <p>
          This is a controlled drawer
        </p>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        <div
          style="display: flex; justify-content: flex-start;"
        >
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary t-drawer__confirm"
            type="button"
          >
            <span
              class="t-button__text"
            >
              确认
            </span>
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default t-drawer__cancel"
            type="button"
          >
            <span
              class="t-button__text"
            >
              取消
            </span>
          </button>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       打开抽屉 
      
    </span>
  </button>
</div>
`;

exports[`Drawer Drawer customVue demo works fine 1`] = `
<div>
  <!-- 使用插槽自定义 -->
  <div
    class="t-drawer t-drawer--right"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        
        <div>
          自定义头部
        </div>
        
      </div>
      
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <p>
          This is the body of drawer
        </p>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        
        <button
          class="t-button t-size-m t-button--variant-base t-button--theme-primary"
          type="button"
        >
          <span
            class="t-button__text"
          >
            
            确定
            
          </span>
        </button>
        <button
          class="t-button t-size-m t-button--variant-outline t-button--theme-default"
          type="button"
        >
          <span
            class="t-button__text"
          >
            
             取消 
            
          </span>
        </button>
        
      </div>
    </div>
  </div>
  <!-- 使用 props 自定义 -->
  <div
    class="t-drawer t-drawer--right"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        抽屉标题
      </div>
      
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <p>
          This is the body of drawer
        </p>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        <div>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary"
            type="button"
          >
            <span
              class="t-button__text"
            >
              
              confirm
              
            </span>
          </button>
          <button
            class="t-button t-size-m t-button--variant-outline t-button--theme-default"
            type="button"
          >
            <span
              class="t-button__text"
            >
              
              cancel
              
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <!-- 单独定义确认/取消按钮 -->
  <div
    class="t-drawer t-drawer--right"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        I am the title
      </div>
      
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <p>
          This is the body of drawer
        </p>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        <div
          style="display: flex; justify-content: flex-start;"
        >
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary t-is-disabled t-drawer__confirm"
            disabled=""
            type="button"
          >
            <span
              class="t-button__text"
            >
              OK
            </span>
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default"
            type="button"
          >
            <span
              class="t-button__text"
            >
              cancel
            </span>
          </button>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       Open(插槽定义) 
      
    </span>
  </button>
      
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       Open(属性定义) 
      
    </span>
  </button>
      
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       Open(单独设置确认/取消按钮) 
      
    </span>
  </button>
</div>
`;

exports[`Drawer Drawer destroyVue demo works fine 1`] = `
<div>
  <!-- 动画效果待开发 -->
  <!---->
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
      打开抽屉
      
    </span>
  </button>
</div>
`;

exports[`Drawer Drawer noMaskVue demo works fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer--right t-drawer--without-mask"
  >
    <!---->
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        抽屉标题
      </div>
      <!---->
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <p>
          抽屉的内容
        </p>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        <div
          style="display: flex; justify-content: flex-start;"
        >
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary t-drawer__confirm"
            type="button"
          >
            <span
              class="t-button__text"
            >
              确认
            </span>
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default t-drawer__cancel"
            type="button"
          >
            <span
              class="t-button__text"
            >
              取消
            </span>
          </button>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       打开抽屉 
      
    </span>
  </button>
</div>
`;

exports[`Drawer Drawer operationVue demo works fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer--right"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        标题名称
      </div>
      <!---->
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <div
          class="t-drawer-demo-div"
        >
          <span>
            标签 A
          </span>
          <div
            class="t-input__wrap"
          >
            <div
              class="t-input"
            >
              <!---->
              <!---->
              <input
                class="t-input__inner"
                placeholder="请输入"
                type="text"
              />
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
          </div>
        </div>
        <div
          class="t-drawer-demo-div"
        >
          <span>
            标签 B
          </span>
          <div
            class="t-input__wrap"
          >
            <div
              class="t-input"
            >
              <!---->
              <!---->
              <input
                class="t-input__inner"
                placeholder="请输入"
                type="text"
              />
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
          </div>
        </div>
        <div
          class="t-drawer-demo-div"
        >
          <span>
            标签 C
          </span>
          <div
            class="t-input__wrap"
          >
            <div
              class="t-input"
            >
              <!---->
              <!---->
              <input
                class="t-input__inner"
                placeholder="请输入"
                type="text"
              />
              <!---->
              <!---->
              <!---->
            </div>
            <!---->
          </div>
        </div>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        <div
          style="display: flex; justify-content: flex-start;"
        >
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary t-drawer__confirm"
            type="button"
          >
            <span
              class="t-button__text"
            >
              确认
            </span>
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default t-drawer__cancel"
            type="button"
          >
            <span
              class="t-button__text"
            >
              取消
            </span>
          </button>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       打开抽屉 
      
    </span>
  </button>
</div>
`;

exports[`Drawer Drawer placementVue demo works fine 1`] = `
<div>
  <!-- @click-cancel 和 :onClickCancel 两种写法都支持； :onClickOverlay 和 @click-overlay 两种写法都支持-->
  <div
    class="t-drawer t-drawer--right"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        标题名称
      </div>
      <!---->
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <p>
          抽屉的内容
        </p>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        <div
          style="display: flex; justify-content: flex-start;"
        >
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary t-drawer__confirm"
            type="button"
          >
            <span
              class="t-button__text"
            >
              确认
            </span>
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default t-drawer__cancel"
            type="button"
          >
            <span
              class="t-button__text"
            >
              取消
            </span>
          </button>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div
    class="tdesign-radio-button"
  >
    <div
      class="t-radio-group t-size-m t-radio-group__outline"
    >
      
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="left"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           左侧 
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="right"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           右侧 
          
          
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="top"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           上方 
          
          
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="bottom"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           下方 
          
          
        </span>
      </label>
      
      <!---->
    </div>
  </div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default btn-top-margin"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       打开抽屉 
      
    </span>
  </button>
</div>
`;

exports[`Drawer Drawer popupVue demo works fine 1`] = `
<div>
  <div
    class="tdesign-radio-button"
  >
     抽屉弹出模式： 
    <div
      class="t-radio-group t-size-m t-radio-group__outline"
    >
      
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="overlay"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           overlay 
          
          
        </span>
      </label>
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="push"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           push 
          
          
        </span>
      </label>
      
      <!---->
    </div>
  </div>
  <br />
  <br />
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
      打开抽屉
      
    </span>
  </button>
</div>
`;

exports[`Drawer Drawer sizeVue demo works fine 1`] = `
<div>
  <div
    class="t-drawer t-drawer--right"
  >
    <div
      class="t-drawer__mask"
    />
    <div
      class="t-drawer__content-wrapper t-drawer__content-wrapper--right"
      style="width: 300px;"
    >
      <div
        class="t-drawer__header"
      >
        抽屉标题
      </div>
      <!---->
      <div
        class="t-drawer__body narrow-scrollbar"
      >
        
        <p>
          抽屉的内容
        </p>
        
      </div>
      <div
        class="t-drawer__footer"
      >
        <div
          style="display: flex; justify-content: flex-start;"
        >
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-primary t-drawer__confirm"
            type="button"
          >
            <span
              class="t-button__text"
            >
              确认
            </span>
          </button>
          <button
            class="t-button t-size-m t-button--variant-base t-button--theme-default t-drawer__cancel"
            type="button"
          >
            <span
              class="t-button__text"
            >
              取消
            </span>
          </button>
          <!---->
        </div>
      </div>
    </div>
  </div>
  <div
    class="tdesign-radio-button"
  >
    <div
      class="t-radio-group t-size-m t-radio-group__outline btn-top-margin"
    >
      
      <label
        class="t-radio-button t-is-checked"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="small"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           小 
          
          
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="medium"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           中 
          
          
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="large"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           大 
          
          
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="200"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           200 
          
          
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="400px"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           400px 
          
          
        </span>
      </label>
      <label
        class="t-radio-button"
      >
        <input
          class="t-radio-button__former"
          name=""
          type="radio"
          value="50%"
        />
        <span
          class="t-radio-button__input"
        />
        <span
          class="t-radio-button__label"
        >
          
          
           50% 
          
          
        </span>
      </label>
      
      <!---->
    </div>
  </div>
  <button
    class="t-button t-size-m t-button--variant-outline t-button--theme-default btn-top-margin"
    type="button"
  >
    <span
      class="t-button__text"
    >
      
       打开抽屉 
      
    </span>
  </button>
</div>
`;
